### PR TITLE
Limb-based damage descriptions fixes and code improvement

### PR DESCRIPTION
**Pull request by FernandoJ8**
## About The Pull Request

The limb-based damage descriptions added by #71635 didn't work properly, as get_majority_bodypart_damage_desc() didn't return the correct values.
I rewrote the proc altogether and changed the format of the damage description vars (formerly burn_damage_desc and brute_damage_desc). They are now a single associative list with each description being indexed by it's respective damage type DEFINE
The proc now returns the correct description, avoids the duplicated code between the brute and burn sections and can support any number of damage types. To that end I also added Clone damage support, although no limbs use a special description for the moment.

Oh, I also gave the robotic damage descriptions to all the robotic limbs, for some reason only the left robot arm had it.
## Why It's Good For The Game

It makes the system function properly and correctly labels robotic limbs.
On the side of the proc, it's more flexible this way and avoids the redundancy. I am aware, however, that I'm using a lot of associative lists. I'll rewrite it to reduce them if memory is deemed significant concern in this context. It's, admittedly, probably a bit overengineered for what it is.
## Changelog
:cl:
fix: the examine text for damage now properly uses the most common examine text out of all the mob's limbs
fix: all robotic limbs now have the robotic damage descriptions (charring, denting)
code: improved the get_majority_bodypart_damage_desc() proc to reduce redundancies and add support for limb-based cellular damage
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_helpers.dm [hunk 1]
```diff
@@ -7,28 +7,24 @@
...
-			seen_burn[part.burn_damage_desc] = 1
-		else
-			seen_burn[part.burn_damage_desc] += 1
-		if(seen_burn[part.burn_damage_desc] > most_seen_burn)
-			burn_desc = part.burn_damage_desc
-	return list(BRUTE = brute_desc, BURN = burn_desc)
+		for(var/i in 1 to part.damage_examines.len)
+			var/damage_type = part.damage_examines[i]
+			var/damage_desc = part.damage_examines[damage_type]
+			if(!seen_damage[damage_type])
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/carbon/human/human_helpers.dm:17]: ``seen_damage`` is never populated with anything, it's always just ``list()``, is this intentional?

Comment (FernandoJ8) [reply code/modules/mob/living/carbon/human/human_helpers.dm:17]: Fixed in the new commit, thank you.

File: code/modules/mob/living/carbon/human/human_helpers.dm [hunk 2]
```diff
@@ -7,28 +7,25 @@
...
-			brute_desc = part.brute_damage_desc
-		//burn
-		if(!seen_burn[part.burn_damage_desc])
-			seen_burn[part.burn_damage_desc] = 1
-		else
-			seen_burn[part.burn_damage_desc] += 1
-		if(seen_burn[part.burn_damage_desc] > most_seen_burn)
-			burn_desc = part.burn_damage_desc
-	return list(BRUTE = brute_desc, BURN = burn_desc)
+		for(var/i in 1 to part.damage_examines.len)
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/human_helpers.dm:14]: Simple in loop here, you don't really need the index ? ```for(var/damage_type in part.damage_examines)```

Comment (FernandoJ8) [reply code/modules/mob/living/carbon/human/human_helpers.dm:14]: I need the index (aka, damage type) to compare the damage descriptions to only the damage descriptions of that same type.

Comment (jlsnow301) [reply code/modules/mob/living/carbon/human/human_helpers.dm:14]: you should still be able to make this comparison, I'm confused, it's just saving a line

Comment (FernandoJ8) [reply code/modules/mob/living/carbon/human/human_helpers.dm:14]: > you should still be able to make this comparison, I'm confused, it's just saving a line

It's entirely possible that I'm being thick in this moment, but how exactly would I make the comparison without having the index?

Comment (jlsnow301) [reply code/modules/mob/living/carbon/human/human_helpers.dm:14]: likewise, I may be too. I don't see anywhere else you need the index. The code could be 
```
for(var/damage_type in part.damage_examines)
    var/damage_desc = part.damage_examines[damage_type]
```

Comment (FernandoJ8) [reply code/modules/mob/living/carbon/human/human_helpers.dm:14]: > likewise, I may be too. I don't see anywhere else you need the index. The code could be
> 
> ```
> for(var/damage_type in part.damage_examines)
>     var/damage_desc = part.damage_examines[damage_type]
> ```

Huh, I thought that would cycle through the description texts themselves rather the indices.
Turns out, I am an idiot.
Thank you.

**Comment by jlsnow301** [comment]
@FernandoJ8 is this finished?

**Pull request merged**